File: src/algorithms/strings/palindrome-check/solution1.go
package main

// O(n^2) time | O(n) space
func IsPalindrome(str string) bool {
	reversed := ""
	for i := len(str) - 1; i >= 0; i-- {
		reversed += string(str[i])
	}
	for i := range str {
		if reversed[i] != str[i] {
			return false
		}
	}
	return true
}

File: src/algorithms/strings/palindrome-check/solution1.ts
// O(n^2) time | O(n) space
export function isPalindrome(string: string) {
  let reversedString = "";
  for (let i = string.length - 1; i >= 0; i--) {
    reversedString += string[i];
  }
  return string === reversedString;
}
